### PR TITLE
AWS ECS Detector: set cloud.{account.id,availability_zone,region}

### PR DESCRIPTION
**Pull request by mmanciop**
**Description:**

Add support in the AWS ECS detector to collect the `cloud.account.id`, `cloud.availability_zone`, `cloud.region` and `cloud.resource_id` (as discussed in https://github.com/open-telemetry/semantic-conventions/issues/677) in line with what the AWS EC2 detector already supports.

**Existing Issue(s):**

N/A

**Testing:**

Unit tests updated. Test data was already sufficient.

**Documentation:**

N/A

**Outstanding items:**

N/A

### CODE REVIEW

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java [hunk 1]
```diff
@@ -99,6 +99,24 @@ static void fetchMetadata(
     }
   }
 
+  @Nullable
+  private static String getAccountId(@Nullable String arn) {
+    if (arn != null) {
+      return arn.split(":")[4];
```
Comment (zeitlinger) [inline aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:105]: avoid index out of bounds if data is invalid

Comment (mmanciop) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java]: Done in d6dd717

**Review by zeitlinger — APPROVED** [review]
great :smile:

**Comment by trask** [comment]
cc @srprash @wangzlei

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java [hunk 2]
```diff
@@ -99,6 +99,31 @@ static void fetchMetadata(
...
+  }
+
+  @Nullable
+  private static String getRegion(@Nullable String arn) {
+    return getArnPart(arn, 3);
+  }
+
+  @Nullable
+  private static String getArnPart(@Nullable String arn, int partIndex) {
+    if (arn == null) {
```
Comment (breedx-splk) [inline aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:114]:
```suggestion
    if (arn == null || partIndex < 0) {
```

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java [hunk 3]
```diff
@@ -146,6 +182,12 @@ static void parseResponse(
           logArnBuilder.setRegion(value);
           break;
         case "TaskARN":
+          if (accountId == null) {
+            accountId = getAccountId(value);
+          }
+          if (region == null) {
+            region = getRegion(value);
+          }
```
Comment (breedx-splk) [inline aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:190]: This is duplicated above, can you consolidate?

**Review by breedx-splk — APPROVED** [review]

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java (hunk 2 above)
Comment (mmanciop) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:114]: Pretty overkill if you ask me, but I can make a private enum out of it

Comment (mmanciop) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:124]: Introduced enum for ArnParts in 367c715

Comment (breedx-splk) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:122]: Thanks, and I did agree that it's a little silly. :) Appreciate the extra mile tho.

**Review by breedx-splk — APPROVED** [review]

**Comment by trask** [comment]
@srprash @wangzlei please review when you have a chance, thx!

**Comment by trask** [comment]
@srprash @wangzlei ping

**Pull request merged**